src/main/crc.ts

```diff
@@ -0,0 +1,54 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { crc32 } from "crc";
+import * as yauzl from "yauzl";
+
+type TCrcFile = [string, string];
+
+export async function extractCrc32OnZip(filePath: string) {
+    const fileArray: TCrcFile[] = [];
+    return new Promise<string>((resolve, reject) => {
+
+        yauzl.open(filePath, { lazyEntries: true }, (err, data) => {
+            if (err) {
+                reject(err);
+            }
+            data.on("error", (e) => reject(e));
+            data.readEntry();
+            data.on("entry", (entry: yauzl.Entry) => {
```

Comment:
Note that `entry` can be a folder (no CRC), so should preferably be skipped with  `entry.fileName[entry.fileName.length - 1] === "/"` (because `entry.crc32` can actually be the number `0`, not just `null` or `undefined`)

Reply:
I prefer to take the directories, they will be counted in the checksum. If a empty directory is added in zip the checksum will be different

What do you think ?

Reply:
Good point!